core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1806,6 +1816,181 @@ public void testShareAutoOffsetResetByDurationInvalidFormat() throws Exception {
         }
     }
 
+    @ClusterTest(
+        brokers = 3,
+        serverProperties = {
+            @ClusterConfigProperty(key = "auto.create.topics.enable", value = "false"),
+            @ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "classic,consumer,share"),
+            @ClusterConfigProperty(key = "group.share.enable", value = "true"),
+            @ClusterConfigProperty(key = "group.share.partition.max.record.locks", value = "10000"),
+            @ClusterConfigProperty(key = "group.share.record.lock.duration.ms", value = "15000"),
+            @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "3"),
+            @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "3"),
+            @ClusterConfigProperty(key = "share.coordinator.state.topic.min.isr", value = "1"),
+            @ClusterConfigProperty(key = "share.coordinator.state.topic.num.partitions", value = "3"),
+            @ClusterConfigProperty(key = "share.coordinator.state.topic.replication.factor", value = "3"),
+            @ClusterConfigProperty(key = "transaction.state.log.min.isr", value = "1"),
+            @ClusterConfigProperty(key = "transaction.state.log.replication.factor", value = "3"),
+            @ClusterConfigProperty(key = "unstable.api.versions.enable", value = "true")
+        }
+    )
+    public void testShareConsumerAfterCoordinatorMovement() throws Exception {
+        setup();
+        String topicName = "multipart";
+        String groupId = "multipartGrp";
+        createTopic(topicName, 3, 3);
+
+        try (Admin admin = createAdminClient()) {
+            TopicPartition tpMulti = new TopicPartition(topicName, 0);
+
+            // get topic id
+            Uuid topicId = admin.describeTopics(List.of(topicName)).topicNameValues().get(topicName).get().topicId();
+
+            // produce some messages
+            try (Producer<byte[], byte[]> producer = createProducer()) {
+                ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+                    tpMulti.topic(),
+                    tpMulti.partition(),
+                    null,
+                    "key".getBytes(),
+                    "value".getBytes()
+                );
+                IntStream.range(0, 10).forEach(__ -> producer.send(record));
+                producer.flush();
+            }
+
+            // consume messages
+            try (ShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(groupId)) {
+                shareConsumer.subscribe(List.of(topicName));
+                alterShareAutoOffsetReset(groupId, "earliest");
```

Comment:
Experience with other tests tells us that it's best to set the auto-offset reset earlier in the test. It's an asynchronous action that may or may not take effect by the time consumption begins.